staging/src/k8s.io/client-go/tools/cache/listers_test.go

```diff
@@ -113,12 +115,17 @@ func TestGenericListerListMethod(t *testing.T) {
 	target := NewGenericLister(indexer, schema.GroupResource{Group: "group", Resource: "resource"})
 
 	expectedOutput := []runtime.Object{
+		newUnstructured("group/version", "TheKind", "ns-bar", "name-bar"),
 		newUnstructured("group/version", "TheKind", "ns-foo", "name-foo"),
 		newUnstructured("group/version", "TheKind", "ns-foo", "name-foo1"),
-		newUnstructured("group/version", "TheKind", "ns-bar", "name-bar"),
 	}
 	actualOutput, err := target.List(labels.Everything())
 
+	// Lister result is unsorted. Sort before comparing result with expected.
+	sort.Slice(actualOutput, func(i, j int) bool {
+		return actualOutput[i].(*unstructured.Unstructured).GetName() < actualOutput[j].(*unstructured.Unstructured).GetName()
```

Comment:
This cheats with namespace, sorting should include namespace, just in this case order from namespace is same as order from name.

Reply:
Prepared https://github.com/kubernetes/kubernetes/pull/130746 as I knew implementing sorting would not be pleasant.